pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/SVScanDocIdIterator.java

```diff
@@ -278,7 +278,8 @@ public boolean doesValueMatch(int docId) {
     public int matchValues(int limit, int[] docIds) {
       _reader.readDictIds(docIds, limit, _buffer, _readerContext);
       int newLimit = MatcherUtils.removeNullDocs(docIds, _buffer, limit, _nullBitmap);
-      assert newLimit < limit;
+      // docIds might not be all documents, so there is no guarantee there are null docs in the docIds.
+      assert newLimit <= limit;
```

Comment:
We can probably remove this assert because I don't think it helps with readability or debugging